assets/js/base/context/providers/cart-checkout/payment-methods/use-payment-method-registration.ts

```diff
@@ -68,6 +67,7 @@ const usePaymentMethodRegistration = (
 
 	useEffect( () => {
 		canPayArgument.current = {
+			cart,
```

Comment:
How about spreading cart here, which would include cartTotals, cartNeedsShipping, paymentRequirements, and also add the rest of items at the same level?

Reply:
I would advise against it, as `cart` has [a lot of keys](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/7595c3374dddeba8e201f159d9a37c09ae27b309/assets/js/types/type-defs/cart.ts#L173-L189), and that would make the signature really confusing. After deprecating the keys you mentioned it would just be `cart` and `billingData` and that makes it easier to understand where the data comes from.

Reply:
Wouldn't having cart as is just complicated one step further? so instead of `arg.cartTotals` you need `arg.cart.cartTotals` especially seeing that canPayArgument is made of cart related fields only.

Reply:
It's also made of `billingData` and `shipping data` from the `CustomerDataContext`

Reply:
Those are also present in Cart, shipping data is `shippingAddress` and billing data is `billingAddress`.

The only difference here is that `billingAddress` reflects what's currently in state, and what's in `Cart` reflects what comes from the server, so this might not always be in sync. billingData can still replace billingAddress from cart.

This is not a huge point to block the PR on as moving the other direction (spreading) is easier than reversing.

:shipit:

Reply:
The data is indeed out of sync. This would be an issue for extensions that need to make calculations on the current `billingData`.

> This is not a huge point to block the PR on as moving the other direction (spreading) is easier than reversing.
👍